action.yml

```diff
@@ -134,39 +134,93 @@ runs:
         app_id: ${{ env.APP_ID }}
         private_key: ${{ env.APP_PRIVATE_KEY }}
 
-    - name: Commit and Push changes
+    - name: Install dependencies for GitHub Auth
+      shell: bash
+      run: |
+        yarn add "@actions/github"
+
+    - name: Update manifest.json and dist folder
       shell: bash
       env:
         GITHUB_TOKEN: ${{ steps.get_installation_token.outputs.token || github.token }}
       run: |
-        app_token="${{ steps.get_installation_token.outputs.token }}" # Assuming this outputs your app token
-        
-        user_info=$(curl -s -H "Authorization: token $app_token" https://api.github.com/user)
-        
-        user_name=$(echo "$user_info" | jq -r .name)
-        user_email=$(echo "$user_info" | jq -r .email)
-        
-        if [ "$user_email" == "null" ]; then
-          user_email="ubiquity-os[bot]@users.noreply.github.com"
-          user_name="ubiquity-os[bot]"
-        fi
-        
-        # Configure Git with the retrieved user information
-        git config --global user.name "$user_name"
-        git config --global user.email "$user_email"
-        git remote set-url origin https://${{ env.GITHUB_TOKEN }}@github.com/${{ github.repository }}.git
-        
         git add "${{ inputs.manifestPath }}"
         git add -f ${{ github.workspace }}/dist/\*
+        echo "Changed files:"
+        echo "$(git diff-index --cached --name-only HEAD)"
         if [ -n "$(git diff-index --cached --name-only HEAD)" ]; then
-          git commit -m "${{ inputs.commitMessage }}" || echo "Commit failed"
-          # Attempt to pull and resolve conflicts
-          git pull origin ${{ github.ref_name }} || {
-            echo "Rebase failed, force pushing changes."
-            git push --force origin HEAD:${{ github.ref_name }}
-            exit 0
-          }
-          git push origin HEAD:${{ github.ref_name }}
+          node -e "
```

Comment:
This is an interesting technique

Reply:
It was the easiest to avoid having to build a tree and comparing SHA for each file by inlining the code.